src/rgw/rgw_rest_pubsub.cc

```diff
@@ -195,13 +195,17 @@ class RGWPSCreateTopicOp : public RGWOp {
       return 0;
     }
     if (ret == 0) {
-      if (result.user == s->owner.id ||
-          !s->cct->_conf->mandatory_topic_permissions) {
+      ret = verify_topic_owner_or_policy(
+          s, result, driver->get_zone()->get_zonegroup().get_name(),
+          rgw::IAM::snsGetTopicAttributes);
+      if (ret == 0)
+      {
         return 0;
       }
-      ldpp_dout(this, 1) << "failed to create topic '" << topic_name
+
+      ldpp_dout(this, 1) << "no permission to modify topic '" << topic_name
                          << "', topic already exist." << dendl;
-      return -EPERM;
+      return -EACCES;
```

Comment:
one thing i really liked about @qiuxinyidian's original change is that it returned a specific error for this case. while that error code `TopicExistsInAnotherUser` doesn't seem to exist in aws, it does clarify why the access was denied in a way that's visible to the client. i think that's important since we're changing the default behavior here

instead of changing that Code, what do you think about adding a Message to the error reply? for example:
```c++
s->err.message = "Topic was already created by another user.";
return -EACCES;
```